contracts/HeapOrdering.sol

```diff
@@ -141,18 +141,22 @@ library HeapOrdering {
         Account memory accountToShift = getAccount(_heap, _rank);
         uint256 valueToShift = accountToShift.value;
         Account memory childToSwap;
+        Account memory rightChild;
         uint256 childRank = _rank << 1;
         // At this point, childRank (resp. childRank+1) is the rank of the left (resp. right) child.
 
         while (childRank <= size) {
-            // Compute the rank of the child with largest value.
-            if (
-                childRank < size &&
-                getAccount(_heap, childRank + 1).value > getAccount(_heap, childRank).value
-            ) childRank++;
-
             childToSwap = getAccount(_heap, childRank);
 
+            // Find the child with largest value.
+            if (childRank < size) {
+                rightChild = getAccount(_heap, childRank + 1);
+                if (rightChild.value > childToSwap.value) {
+                    childRank++;
```

Comment:
```suggestion
                    unchecked {
                        ++childRank;
                    }
```

to save gas

Reply:
This variable is multiplied by 2 many times, so I don't think it applies safely here (even if it probably does, because childrank is related to a rank in an array). It is completely safe for variables that are only incremented, because you know you won't ever be able to reach 2^256 by only incrementing

Reply:
After talking with @MathisGD, I think this is safe actually because we know that `childRank < size`. So it can't overflow